boot/boot.go

```diff
@@ -360,5 +358,35 @@ func MakeBootable(model *asserts.Model, rootdir string, bootWith *BootableSet) e
 	}
 
 	return nil
+}
+
+func makeBootable20(model *asserts.Model, rootdir string, bootWith *BootableSet, opts *bootloader.Options) error {
+	// install the bootloader configuration from the gadget
+	if err := bootloader.InstallBootConfig(bootWith.UnpackedGadgetDir, rootdir, opts); err != nil {
+		return err
+	}
+
+	// XXX: extract kernel for e.g. ARM
+	//
+	// XXX2: "pseudo" symlink kernel to /systems/XXXX/kernel and add
```

Comment:
Need to double check there is only a single recovery system at this point (because grub will not know what to boot otherwise). Also needs to create this pseudo symlink *or* grubenv with just a single snapd_recovery_kernel= variable.

Reply:
I added a check to ensure that there are not multiple recovery systems on disk now. In https://github.com/snapcore/snapd/compare/master...mvo5:boot-uc20-5?expand=1 I have code now (specifically in https://github.com/snapcore/snapd/commit/2b051597e2d2ec305293dccf7ffce844530a2660) that highlights how the "snapd_recovery_kernel" variable is set via a recovery system specific bootenv.